### PR TITLE
Sorts : Added Block Sort Algorithm

### PR DESCRIPTION
**Pull request by SportingDot1024**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by Aditya-K-Pathak — APPROVED** [review]

File: sorts/block_sort.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return []
+
+    block_size = int(len(lst) ** 0.5)
+    blocks = [lst[i : i + block_size] for i in range(0, len(lst), block_size)]
+    for block in blocks:
+        block.sort()
+
+    sorted_lst = []
+    while blocks:
+        min_block = min(blocks, key=lambda x: x[0])
```
[the remark this answers is not in the document]
Comment (SportingDot1024) [reply sorts/block_sort.py]: done

**Review by Aditya-K-Pathak — APPROVED** [review]
Add Comments for understanding if possible, else it's fine

**Review by Aditya-K-Pathak — CHANGES_REQUESTED** [review]
Line 1: use list instead of List
Line 41 is too long, break it from comma and use a full stop their and start the rest in new line with caps
Line 45 return sorted result, you are storing result in sorted_result not in sorted_lst

**Comment by SportingDot1024** [comment]
I have added comments.

**Comment by SportingDot1024** [comment]
All the checks have been passed.

**Comment by SportingDot1024** [comment]
#10490

**Pull request closed without merging**